modules/libcom/src/iocsh/iocsh.cpp

```diff
@@ -545,6 +546,20 @@ int iocshSetError(int err)
     return err;
 }
 
+void addPwdMacro(iocshContext *context) {
+    if (context != NULL) {
```

Comment:
fyi. `context` can never be null where this function is (or should) be called.

Reply:
I can't assume that this function will ever be called from this single point in code in the future. Don't you think it is better to prevent now?

Reply:
No code is fully safe from future regressions.  Personally, I don't think I would have bothered.  Or at most `assert(context);`. Having a function `addPwdMacro()` which can silently not add the PWD macro is also an avenue for future surprise.

Reply:
Ok, I'll study better how this context structure works. With your two comments about it, I think I'm missing an important point.

Reply:
We can talk though this if you like.

Reply:
Thank you. I usually need a moment alone with the code to figure things out. But I'll reach to you if I see it is taking too long.